shhext/service_test.go

```diff
@@ -0,0 +1,144 @@
+package shhext
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestShhExtSuite(t *testing.T) {
+	suite.Run(t, new(ShhExtSuite))
+}
+
+type ShhExtSuite struct {
+	suite.Suite
+
+	nodes    []*node.Node
+	services []*Service
+	whisper  []*whisper.Whisper
+}
+
+func (s *ShhExtSuite) SetupTest() {
+	s.nodes = make([]*node.Node, 2)
+	s.services = make([]*Service, 2)
+	s.whisper = make([]*whisper.Whisper, 2)
+	port := 32139
+	for i := range s.nodes {
+		i := i // bind i to be usable in service constructors
+		key, err := crypto.GenerateKey()
+		s.NoError(err)
+		cfg := &node.Config{
+			Name: fmt.Sprintf("node-%d", i),
+			P2P: p2p.Config{
+				NoDiscovery: true,
+				MaxPeers:    20,
+				PrivateKey:  key,
+				ListenAddr:  fmt.Sprintf(":%d", port+i),
```

Comment:
`MaxPeers`, `PrivateKey` and `ListenAddr` may be not needed. I was only using
```go
nodeA, nodeErr := node.New(&node.Config{NoUSB: true})
```
and it worked fine.

Reply:
MaxPeers and ListenAddr turned out to be required, otherwise, peers can't connect in my case

looks like PrivateKey is generated if not provided

Reply:
Just verified that and you're right. But in my case, `ListenAddr: ":0"` did a trick as the OS binds a free port.